frontend/src/metabase/search/components/SearchFilterModal/filters/TypeFilter.tsx

```diff
@@ -0,0 +1,47 @@
+/* eslint-disable react/prop-types */
+import { t } from "ttag";
+import { getTranslatedEntityName } from "metabase/nav/utils";
+import { Checkbox } from "metabase/ui";
+import { useSearchListQuery } from "metabase/common/hooks";
+import LoadingSpinner from "metabase/components/LoadingSpinner";
+import { SearchFilter } from "metabase/search/components/SearchFilterModal/filters/SearchFilter";
+
+import { SearchFilterComponent } from "metabase/search/util/filter-types";
+import { TypeCheckboxGroupWrapper } from "metabase/search/components/SearchFilterModal/filters/TypeFilter.styled";
+
+const EMPTY_SEARCH_QUERY = { models: "dataset", limit: 1 } as const;
+
+export const TypeFilter: SearchFilterComponent<"type"> = ({
+  value = [],
+  onChange,
+  "data-testid": dataTestId,
+}) => {
+  const { metadata, isLoading } = useSearchListQuery({
+    query: EMPTY_SEARCH_QUERY,
+  });
+
+  const availableModels = (metadata && metadata.available_models) ?? [];
+
+  return isLoading ? (
+    <LoadingSpinner />
+  ) : (
+    <SearchFilter data-testid={dataTestId} title={t`Type`}>
```

Comment:
Might even be able to customize the loading state with a `LoadingTypeFilter` component that uses a skeleton, but that's polish for the future